yellow-paper/docs/l1-smart-contracts/frontier.md

```diff
@@ -0,0 +1,158 @@
+---
+title: Frontier Merkle Tree
+---
+
+The Frontier Merkle Tree is an append only Merkle tree that is optimized for minimal storage on chain.
+By storing only the right-most non-empty node at each level of the tree we can always extend the tree with a new leaf or compute the root without needing to store the entire tree.
+We call these values the frontier of the tree.
+If we have the next index to insert at and the current frontier, we have everything we need to extend the tree or compute the root, with much less storage than a full merkle tree.
```

Comment:
```suggestion
If we have the next index to insert at and the current frontier, we have everything we need to extend the tree or compute the root, with much less storage than a full merkle tree. Note that we're not actually keeping track of the data in the tree: we only store what's minimally required in order to be able to compute the root after inserting a new element.
```

Reply:
It took me a while to understand what the goal of this construct was. I think this early clarification better prepares the reader for what comes next.